velox/common/encode/Base64.cpp

```diff
@@ -81,13 +87,19 @@ constexpr const Base64::ReverseIndex kBase64UrlReverseIndexTable = {
     255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
     255};
 
-constexpr bool checkForwardIndex(
+// Validate the character in charset with ReverseIndex table
+static constexpr bool checkForwardIndex(
     uint8_t idx,
     const Base64::Charset& charset,
-    const Base64::ReverseIndex& table) {
-  return (table[static_cast<uint8_t>(charset[idx])] == idx) &&
-      (idx > 0 ? checkForwardIndex(idx - 1, charset, table) : true);
+    const Base64::ReverseIndex& reverseIndex) {
+  for (uint8_t i = 0; i <= idx; ++i) {
```

Comment:
The recursive algorithm starts from idx down to 0. Let's follow the same to keep it consistent.

Reply:
updated the code